qiskit/algorithms/amplitude_amplifiers/grover.py

```diff
@@ -105,7 +102,7 @@ class Grover(AmplitudeAmplifier):
     """
 
     def __init__(self,
-                 iterations: Optional[Union[int, List[int], Iterator[int], float]] = None,
+                 iterations: Optional[Union[List[int], Iterator[int], float]] = None,
```

Comment:
In this case whether the number is an int or a float type changes behavior, such that passing 1.0 is not the same as passing 1. Often the number will just be used so its type is less important as such, so saying complex and passing a float or an int, as permitted/implied by the typehint will end being the same result. In saying this I see the doctring below does not mention the behavior when a float is passed. @Cryoris any thoughts here

Reply:
Good catch. I  do not like  this behaviour. Since this is the first release under terra we have a chance to slightly change the API and drop the int special case. Its easy enough for users to write `[iterations]` instead of `iterations` if that's their intention.

Reply:
@Cryoris says that float was a holdover from previous attempt and can be dropped.I will update